Long database migrations fail in Kubernetes

### Description

Long running database migrations ran in a container can fail after a long period of time with `An I/O error occurred while sending to the backend.` It seems that the underlying database connection or socket is being closed by HikariCP, CNI, OS, etc. while the migration is still running. Perhaps they detect a lack of activity on the connection and kill it. Flyway is unable to handle this and can't even cleanup gracefully since without the connection it can't rollback or release its advisory locks.

See also:

https://docs.actian.com/avalanche/index.html#page/User/Network_Disconnects_Due_to_Perceived_Inactivity.htm

### Steps to reproduce

Create a migration with `pg_sleep(600)` and deploy the importer using Helm

### Additional context

```
2021-11-22T10:18:52.206-0600 INFO main o.f.c.i.c.DbMigrate Current version of schema "public": 1.45.1
2021-11-22T10:18:52.223-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.46.0 - entity timestamp"
2021-11-22T10:24:38.439-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.46.1 - revert v1.43.2"
2021-11-22T10:24:38.486-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.46.2 - entity history"
2021-11-22T10:24:55.769-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.46.3 - smart contracts"
2021-11-22T10:25:46.277-0600 INFO main o.f.c.i.c.DbMigrate Migrating schema "public" to version "1.46.5 - combine topic id"
2021-11-23T03:29:55.130-0600 ERROR main o.f.c.i.j.TransactionalExecutionTemplate Unable to rollback transaction org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:877)
	at org.postgresql.jdbc.PgConnection.rollback(PgConnection.java:884)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:82)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:328)
	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:291)
	at org.flywaydb.core.internal.command.DbMigrate.access$100(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:195)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:192)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:69)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:192)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:152)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:572)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:32)
2021-11-23T03:29:55.131-0600 ERROR main o.f.c.i.j.TransactionalExecutionTemplate Unable to restore autocommit to original value for connection org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:877)
	at org.postgresql.jdbc.PgConnection.setAutoCommit(PgConnection.java:806)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:97)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:328)
	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:291)
	at org.flywaydb.core.internal.command.DbMigrate.access$100(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:195)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:192)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:69)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:192)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:152)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:572)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:32)
2021-11-23T03:29:55.132-0600 ERROR main o.f.c.i.c.DbMigrate Migration of schema "public" to version "1.46.5 - combine topic id" failed! Changes successfully rolled back.
2021-11-23T03:29:55.132-0600 ERROR main o.f.c.i.d.p.PostgreSQLAdvisoryLockTemplate Unable to release PostgreSQL advisory lock org.postgresql.util.PSQLException: This connection has been closed.
	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:877)
  at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:1734)
	at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:428)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.prepareStatement(JdbcTemplate.java:344)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.queryForBoolean(JdbcTemplate.java:170)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.unlock(PostgreSQLAdvisoryLockTemplate.java:101)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:80)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:192)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:152)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:572)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:32)
2021-11-23T03:29:55.134-0600 WARN main o.s.b.w.r.c.AnnotationConfigReactiveWebServerApplicationContext Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException:
Migration V1.46.5__combine_topic_id.sql failed
----------------------------------------------
SQL State  : 08006
Error Code : 0
Message    : An I/O error occurred while sending to the backend.
Location   : db/migration/v1/V1.46.5__combine_topic_id.sql (/app/resources/db/migration/v1/V1.46.5__combine_topic_id.sql)
Line       : 38
Statement  : -- create updated indexes
create index if not exists topic_message__topic_id_timestamp
on topic_message (topic_id, consensus_timestamp)

2021-11-23T03:29:57.334-0600 ERROR main o.s.b.SpringApplication Application run failed org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'flywayInitializer' defined in class path resource [org/springframework/boot/autoconfigure/flyway/FlywayAutoConfiguration$FlywayConfiguration.class]: Invocation of init method failed; nested exception is org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V1.46.5__combine_topic_id.sql failed
----------------------------------------------
SQL State  : 08006
Error Code : 0
Message    : An I/O error occurred while sending to the backend.
Location   : db/migration/v1/V1.46.5__combine_topic_id.sql (/app/resources/db/migration/v1/V1.46.5__combine_topic_id.sql)
Line       : 38
Statement  : -- create updated indexes
create index if not exists topic_message__topic_id_timestamp
    on topic_message (topic_id, consensus_timestamp)

	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1154)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:908)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:64)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332)
	at com.hedera.mirror.importer.MirrorImporterApplication.main(MirrorImporterApplication.java:32)
Caused by: org.flywaydb.core.internal.command.DbMigrate$FlywayMigrateException: 
Migration V1.46.5__combine_topic_id.sql failed
----------------------------------------------
SQL State  : 08006
Error Code : 0
Message    : An I/O error occurred while sending to the backend.
Location   : db/migration/v1/V1.46.5__combine_topic_id.sql (/app/resources/db/migration/v1/V1.46.5__combine_topic_id.sql)
Line       : 38
Statement  : -- create updated indexes
create index if not exists topic_message__topic_id_timestamp
    on topic_message (topic_id, consensus_timestamp)

	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:427)
	at org.flywaydb.core.internal.command.DbMigrate.access$200(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$3.call(DbMigrate.java:331)
	at org.flywaydb.core.internal.jdbc.TransactionalExecutionTemplate.execute(TransactionalExecutionTemplate.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.applyMigrations(DbMigrate.java:328)
	at org.flywaydb.core.internal.command.DbMigrate.migrateGroup(DbMigrate.java:291)
	at org.flywaydb.core.internal.command.DbMigrate.access$100(DbMigrate.java:56)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:195)
	at org.flywaydb.core.internal.command.DbMigrate$2.call(DbMigrate.java:192)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLAdvisoryLockTemplate.execute(PostgreSQLAdvisoryLockTemplate.java:69)
	at org.flywaydb.core.internal.database.postgresql.PostgreSQLConnection.lock(PostgreSQLConnection.java:99)
	at org.flywaydb.core.internal.schemahistory.JdbcTableSchemaHistory.lock(JdbcTableSchemaHistory.java:141)
	at org.flywaydb.core.internal.command.DbMigrate.migrateAll(DbMigrate.java:192)
	at org.flywaydb.core.internal.command.DbMigrate.migrate(DbMigrate.java:152)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:216)
	at org.flywaydb.core.Flyway$1.execute(Flyway.java:165)
	at org.flywaydb.core.Flyway.execute(Flyway.java:572)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	at org.springframework.boot.autoconfigure.flyway.FlywayMigrationInitializer.afterPropertiesSet(FlywayMigrationInitializer.java:66)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 18 more
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V1.46.5__combine_topic_id.sql failed
----------------------------------------------
SQL State  : 08006
Error Code : 0
Message    : An I/O error occurred while sending
to the backend.
Location   : db/migration/v1/V1.46.5__combine_topic_id.sql (/app/resources/db/migration/v1/V1.46.5__combine_topic_id.sql)
Line       : 38
Statement  : -- create updated indexes
create index if not exists topic_message__topic_id_timestamp
    on topic_message (topic_id, consensus_timestamp)

	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:271)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:218)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:126)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:78)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.lambda$execute$0(SqlMigrationExecutor.java:67)
	at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:27)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:66)
	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:424)
	... 38 more
Caused by: org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:349)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:481)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:401)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:322)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:308)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:284)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:279)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:241)
	at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:111)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:206)
	... 44 more
Caused by: java.io.EOFException
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:445)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2056)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:322)
	... 53 more
```

### Hedera network

mainnet

### Version

v0.43.2

### Operating system

Linux

--- a/hedera-mirror-grpc/src/main/resources/application.yml
+++ b/hedera-mirror-grpc/src/main/resources/application.yml
@@ -19,7 +19,7 @@ logging:
   level:
     root: warn
     com.hedera.mirror.grpc: info
-    org.springframework.cloud.kubernetes.fabric8.config.Fabric8ConfigUtils: error
+    org.springframework.cloud.kubernetes.fabric8.config: error
     # io.grpc: debug
     # org.hibernate.SQL: debug
     # org.hibernate.type.descriptor.sql.BasicBinder: trace
@@ -62,7 +62,7 @@ spring:
   datasource:
     name: ${hedera.mirror.grpc.db.name}
     password: ${hedera.mirror.grpc.db.password}
-    url: jdbc:postgresql://${hedera.mirror.grpc.db.host}:${hedera.mirror.grpc.db.port}/${hedera.mirror.grpc.db.name}?sslmode=${hedera.mirror.grpc.db.sslMode}
+    url: jdbc:postgresql://${hedera.mirror.grpc.db.host}:${hedera.mirror.grpc.db.port}/${hedera.mirror.grpc.db.name}
     username: ${hedera.mirror.grpc.db.username}
     hikari:
       connection-timeout: 3000
@@ -70,6 +70,8 @@ spring:
         idle_in_transaction_session_timeout: "30000"
         lock_timeout: "10000"
         statement_timeout: "60000"
+        sslmode: ${hedera.mirror.grpc.db.sslMode}
+        tcpKeepAlive: true
       maximum-pool-size: 50
       minimum-idle: 4
       validation-timeout: 3000

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImpl.java
@@ -107,6 +107,8 @@ public class AddressBookServiceImpl implements AddressBookService {
             return;
         }
 
+        log.info("Received an address book update: {}", fileData);
+
         // ensure address_book table is populated with latest addressBook prior to additions
         validateAndCompleteAddressBookList(fileData);
 

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/AbstractUpsertQueryGenerator.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/AbstractUpsertQueryGenerator.java
@@ -48,21 +48,14 @@ import org.springframework.util.ReflectionUtils;
 
 @RequiredArgsConstructor
 public abstract class AbstractUpsertQueryGenerator<T> implements UpsertQueryGenerator {
-    protected static final String EMPTY_STRING = "''";
+
+    private static final String EMPTY_STRING = "''";
     private static final String EMPTY_CLAUSE = "";
-    private static final String V1_DIRECTORY = "/v1";
-    private static final String V2_DIRECTORY = "/v2";
     private static final Comparator<DomainField> DOMAIN_FIELD_COMPARATOR = Comparator.comparing(DomainField::getName);
+
     protected final Logger log = LogManager.getLogger(getClass());
-    private final Class<T> metaModelClass = (Class<T>) new TypeToken<T>(getClass()) {
-    }.getRawType();
-    @Getter(lazy = true)
-    private final String insertQuery = generateInsertQuery();
+    private final Class<T> metaModelClass = (Class<T>) new TypeToken<T>(getClass()) { }.getRawType();
     private volatile Set<Field> attributes = null;
-    @Getter(lazy = true)
-    private final String updateQuery = generateUpdateQuery();
-    @Value("${spring.flyway.locations:v1}")
-    private String version;
 
     protected boolean isInsertOnly() {
         return false;
@@ -124,7 +117,8 @@ public abstract class AbstractUpsertQueryGenerator<T> implements UpsertQueryGene
         return Collections.emptySet();
     }
 
-    private String generateInsertQuery() {
+    @Override
+    public String getInsertQuery() {
         StringBuilder insertQueryBuilder = new StringBuilder(StringUtils.joinWith(
                 " ",
                 getCteForInsert(),
@@ -149,7 +143,8 @@ public abstract class AbstractUpsertQueryGenerator<T> implements UpsertQueryGene
         return insertQueryBuilder.toString();
     }
 
-    private String generateUpdateQuery() {
+    @Override
+    public String getUpdateQuery() {
         if (isInsertOnly()) {
             return EMPTY_CLAUSE;
         }

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/UpsertQueryGeneratorFactory.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/UpsertQueryGeneratorFactory.java
@@ -80,7 +80,6 @@ public class UpsertQueryGeneratorFactory {
     }
 
     UpsertEntity createEntity(Class<?> domainClass) {
-        Table table = AnnotationUtils.findAnnotation(domainClass, Table.class);
         Upsertable upsertable = AnnotationUtils.findAnnotation(domainClass, Upsertable.class);
 
         if (upsertable == null) {
@@ -88,6 +87,7 @@ public class UpsertQueryGeneratorFactory {
         }
 
         EntityType<?> entityType = entityManager.getMetamodel().entity(domainClass);
+        Table table = AnnotationUtils.findAnnotation(domainClass, Table.class);
         String tableName = table != null ? table.name() : toSnakeCase(entityType.getName());
         Set<String> idAttributes = getIdAttributes(entityType);
         Set<UpsertColumn> upsertColumns = new TreeSet<>();

--- a/hedera-mirror-importer/src/main/resources/application.yml
+++ b/hedera-mirror-importer/src/main/resources/application.yml
@@ -18,7 +18,7 @@ logging:
     root: warn
     com.hedera.mirror.importer: info
     org.flywaydb.core.internal.command.DbMigrate: info
-    org.springframework.cloud.kubernetes.fabric8.config.Fabric8ConfigUtils: error
+    org.springframework.cloud.kubernetes.fabric8.config: error
     #org.hibernate.type.descriptor.sql.BasicBinder: trace
 management:
   endpoints:
@@ -68,14 +68,15 @@ spring:
   datasource:
     name: ${hedera.mirror.importer.db.name}
     password: ${hedera.mirror.importer.db.password}
-    url: jdbc:postgresql://${hedera.mirror.importer.db.host}:${hedera.mirror.importer.db.port}/${hedera.mirror.importer.db.name}
+    url: jdbc:postgresql://${hedera.mirror.importer.db.host}:${hedera.mirror.importer.db.port}/${hedera.mirror.importer.db.name}?tcpKeepAlive=true
     username: ${hedera.mirror.importer.db.username}
     hikari:
+      # Note: Flyway does not use Hikari so these properties are ignored. Use URL properties for Flyway instead.
       data-source-properties:
         #loggerLevel: TRACE
         logUnclosedConnections: true
         reWriteBatchedInserts: true
-      maximumPoolSize: 14 # recommended connections = ((core_count * 2) + effective_spindle_count), though somewhat outdated as we use SSDs
+      maximumPoolSize: 14
   flyway:
     baselineOnMigrate: true
     connectRetries: 20

--- a/hedera-mirror-monitor/src/main/resources/application.yml
+++ b/hedera-mirror-monitor/src/main/resources/application.yml
@@ -24,7 +24,7 @@ logging:
     com.hedera.mirror.monitor: info
     com.hedera.hashgraph.sdk.Executable: error
     com.hedera.hashgraph.sdk.TransactionReceiptQuery: error
-    org.springframework.cloud.kubernetes.fabric8.config.Fabric8ConfigUtils: error
+    org.springframework.cloud.kubernetes.fabric8.config: error
 management:
   endpoints:
     web:

